### ISSUE TITLE
I made a JSON for use with Optifine (MC 1.14.4)

### ISSUE THREAD

**Issue by sbarrack** [cabaletta/baritone#797]
## Describe your suggestion
So I wanted to use Baritone and Optifine simultaneously and I just copied some of the things from your JSON into my Optifine JSON and it works like a charm! You should totally tell people they can use this.

## Settings
Replace the contents of `.minecraft/versions/1.14.4-OptiFine_HD_U_F3/1.14.4-OptiFine_HD_U_F3.json` with this text. Old Optifine JSON is included at the bottom of this post.

```json
{
  "id": "1.14.4-OptiFine_HD_U_F3",
  "type": "release",
  "inheritsFrom": "1.14.4",
  "time": "2019-07-23T20:50:21Z",
  "releaseTime": "2019-07-23T20:50:21Z",
  "downloads": {},
  "minimumLauncherVersion": 21,
  "mainClass": "net.minecraft.launchwrapper.Launch",
  "arguments": {
    "game": [
     "--tweakClass",
     "baritone.launch.BaritoneTweaker",
     "optifine.OptiFineTweaker"
    ]
  },
  "libraries": [
    {
      "name": "cabaletta:baritone-api:1.4",
      "url": "https://impactdevelopment.github.io/maven/"
    },
    {
      "name": "com.github.ImpactDevelopment:SimpleTweaker:1.2",
      "url": "https://impactdevelopment.github.io/maven/"
    },
    {
      "name": "org.spongepowered:mixin:0.7.11-SNAPSHOT",
      "url": "http://dl.liteloader.com/versions/"
    },
    {
      "name": "org.ow2.asm:asm-all:5.0.3",
      "url": "http://repo1.maven.org/maven2/"
    },
    {
      "name": "optifine:OptiFine:1.14.4_HD_U_F3"
    },
    {
      "name": "optifine:launchwrapper-of:2.1"
    }
  ]
}
```

## Context
This allows the use of both mods at the same time, adding greater compatibility to Baritone. I get no errors when running on setup with Windows 10. I'm not a betting man, but if I did, I'd bet the same procedure works with other launcher mods.

## Final checklist
- [ ] Copy
- [ ] Paste
- [ ] Click
- [ ] Run

[1.14.4-OptiFine_HD_U_F3.json.old.txt](https://github.com/cabaletta/baritone/files/3507387/1.14.4-OptiFine_HD_U_F3.json.old.txt)  

EDIT: There seems to be some confusion about the `.json.old.text`. My phrasing was poor, the attached text file is the old Optifine.json. It does not have the modification. The code above is the modification.

**Comment by leijurv** [comment]
ok this is epic

esaily 500000iq

i think someone else already did this tho, but whatever i'll link to this one

**Comment by RileyGuy** [comment]
I believe this is also possible with MultiMC if you extract the optifine jar into the _MOD file and click the "Add to Minecraft.jar" option in your instance.

**Comment by skyguygamer** [comment]
This Doesnt work because when i do what u say all i get is optifine but theres no baritone

**Comment by RileyGuy** [comment]
> 
> 
> This Doesnt work because when i do what u say all i get is optifine but theres no baritone

Because you need to install baritone, lol.

Either grab it as a forge mod, or use [this](https://github.com/cabaletta/baritone/issues/642#issuecomment-515186974).

**Comment by TheK0tYaRa** [comment]
Currently it gives you 403 error on the optifine link

**Comment by StijnSimons** [comment]
lol cayse you need optifine first??

**Comment by TheK0tYaRa** [comment]
Well, optifine was missing. But Baritone somehow didn't get the problem of "you need to download me". Maybe it's just it getting catched on from EMC

**Comment by leijurv** [comment]
nothing to do with EMC

you may notice that baritone has a URL to download from
```
    {
      "name": "cabaletta:baritone-api:1.4",
      "url": "https://impactdevelopment.github.io/maven/"
    },
```

while optifine does not
```
    {
      "name": "optifine:OptiFine:1.14.4_HD_U_F3"
    },
```

pretty simple lol

**Comment by TheK0tYaRa** [comment]
Well then i don't get the 403 error on the optifine side. but if it works - it works.

**Comment by RileyGuy** [comment]
> 
> 
> Well then i don't get the 403 error on the optifine side. but if it works - it works.

Using the link I sent above, install that custom instance, then extract the optifine jar into it's _MOD equivalent and click "Add to Jar" in your instance settings and add the _MOD file to the jar.

**Comment by hegxig** [comment]
how do u convert the json.old.txt file to a .json file

**Comment by TheK0tYaRa** [comment]
> how do u convert the json.old.txt file to a .json file

Use F2 on the file

**Comment by hegxig** [comment]
![image](https://user-images.githubusercontent.com/34860668/64995396-65de0f80-d8db-11e9-9511-8a2c1c4feb13.png)
what am i doing wrong

**Comment by TheK0tYaRa** [comment]
Download Optifine

**Comment by sbarrack** [comment]
> how do u convert the json.old.txt file to a .json file

@hegxig you rename the file from `.json.old.txt` to `.json`. The content of the file is still JSON despite being a text file.

> Download Optifine, install it and then try replacing the json

@TheK0tYaRa is correct. You don't actually have Optifine installed (no jar in your folder there). Install it first, then navigate to the folder where the jar and json files are contained. Open the existing optifine json and replace the contents with that of the code block in the op. The text file I attached is a backup of the original, not the code needed for the hack.

**Comment by Dazzle20XX** [comment]
So, afterwards do I run the baritone client or optifine client or?

**Comment by sbarrack** [comment]
> So, afterwards do I run the baritone client or optifine client or?

Optifine, you are just replacing the contents of the Optifine.json.

**Comment by Murecore** [comment]
bruh, sorry for bothering but i replaced the content of optifine with this code above but it still didn't work. I tried understanding what you guys did but I have no clue what I'm doing. :(

**Comment by sbarrack** [comment]
> bruh, sorry for bothering but i replaced the content of optifine with this code above but it still didn't work. I tried understanding what you guys did but I have no clue what I'm doing. :(

All I can offer besides make sure you have that version of Opti installed is read the manual. Seems like this is getting a lot of good attention though so maybe I'll do a video tutorial for the bee update.

**Comment by TheK0tYaRa** [comment]
I had to rearrange the parts of the json text to look more like optifine's

**Comment by Ultrabeet1** [comment]
I applied the code to a MC Version file, and then uploaded it so people don't have to do nearly as much work. https://github.com/Ultrabeet1/Baritone-Optifine

**Comment by Lolfaceftw** [comment]
So, what do I edit here when optifine released a new version to 1.14 optifine?

**Comment by S-B99** [comment]
The url.
